dask/_task_spec.py

```diff
@@ -787,6 +790,30 @@ def substitute(
             )
 
 
+class Task(BaseTask):
+    def __setstate__(self, state):
+        (
+            self.key,
+            self.func,
+            self.args,
+            self._token,
+            self._data_producer,
+            self._dependencies,
+            self.kwargs,
+        ) = state
+
+    def __getstate__(self):
+        return (
+            self.key,
+            self.func,
+            self.args,
+            self._token,
+            self._data_producer,
+            self._dependencies,
+            self.kwargs,
+        )
```

Comment:
this is saving a ton when looking at an individual task but I'm not sure if it does when looking at a graph. If it just impacts an individual task this PR may not be worth it (although it still helps a bit when sending stuff to the worker)